### PR TITLE
Mesos docs

### PR DESCRIPTION
**Pull request by djosborne**
Improvements to Mesos docs for simpler, cleaner install story.

Also back applied patch from master to v2.0

### CODE REVIEW

**Comment by robbrockbank** [comment]
A rebase should fix this build failure.  I fixed the broken link earlier today.

File: master/getting-started/mesos/installation/docker.md [hunk 1]
```diff
@@ -5,65 +5,92 @@ title: Installing Calico for the Docker Containerizer
...
 ```shell
-$ docker ps
-CONTAINER ID        NAMES               IMAGE                           CREATED
-f237fb21d357        calico-node         calico/node:latest              3 seconds
+$ curl -o /usr/bin/calicoctl -L http://www.projectcalico.org/builds/calicoctl
+$ chmod a+x calicoctl
 ```
 
-4. Enable Docker Containerizer in Mesos.
+Then, use `calicoctl` to easily launch the `calico/node` container:
```
Comment (robbrockbank) [inline master/getting-started/mesos/installation/docker.md]: remove "easily"

File: master/getting-started/mesos/installation/docker.md [hunk 2]
```diff
@@ -5,65 +5,92 @@ title: Installing Calico for the Docker Containerizer
...
-If using etcd as a cluster store, for example, run the Docker daemon with the
-following additional parameter:
+#### 1. etcd
 
-```shell
-    --cluster-store=etcd://<ETCD HOST>:<PORT>
+Calico uses etcd as its datastore. Ensure you have an instance of etcd running,
+and that it is accessible from all Agents in your cluster.
+
+To maximize availability, we recommend running an etcd cluster across your mesos masters.
```
Comment (robbrockbank) [inline master/getting-started/mesos/installation/docker.md]: Should mesos be Mesos?

File: master/getting-started/mesos/installation/docker.md [hunk 3]
```diff
@@ -5,65 +5,92 @@ title: Installing Calico for the Docker Containerizer
...
+See [etcd's clustering guide](https://coreos.com/os/docs/latest/cluster-architectures.html)
+for more information.
+
+For testing, you can quickly run a single instance of etcd using docker on your Master
```
Comment (robbrockbank) [inline master/getting-started/mesos/installation/docker.md]: remove quickly

File: master/getting-started/mesos/installation/docker.md [hunk 4]
```diff
@@ -5,65 +5,92 @@ title: Installing Calico for the Docker Containerizer
...
-For production deployments, we recommend running the
-container as a service. Visit our guide on [running Calico
-as a service]({{site.baseurl}}/{{page.version}}/usage/configuration/as-service) to learn how to do this.
+By default, Mesos enables on the "Mesos" Containerizer. Ensure
+the Docker Containerizer is also enabled on each Agent.
 
-For test environments that you would like to get up and running
-quickly, you can launch the container with `calicoctl`:
+If you are using the default mesos-slave wrapper from the mesos project,
+you can enable the Docker Containerizer easily with the following command:
```
Comment (robbrockbank) [inline master/getting-started/mesos/installation/docker.md]: remove easily

**Comment by robbrockbank** [comment]
I know it wasn't up for review, but I had a quick read anyways.

File: master/getting-started/mesos/installation/docker.md (hunk 3 above)
Comment (djosborne) [inline master/getting-started/mesos/installation/docker.md]: I'll remove it, but I intentionally put in "quickly" to imply that this is a fast hack, and the clustering is really the correct way to go in prod

File: master/getting-started/mesos/installation/docker.md [hunk 5]
```diff
@@ -5,65 +5,92 @@ title: Installing Calico for the Docker Containerizer
...
-```shell
-    --cluster-store=etcd://<ETCD HOST>:<PORT>
+Calico uses etcd as its datastore. Ensure you have an instance of etcd running,
+and that it is accessible from all Agents in your cluster.
+
+To maximize availability, we recommend running an etcd cluster across your Mesos masters.
+See [etcd's clustering guide](https://coreos.com/os/docs/latest/cluster-architectures.html)
+for more information.
+
+For testing, you can run a single instance of etcd using docker on your Master
```
Comment (robbrockbank) [inline master/getting-started/mesos/installation/docker.md]: I think lower case m for master no?   Or make it upper case everywhere else :-)

File: master/getting-started/mesos/installation/docker.md [hunk 6]
```diff
@@ -5,65 +5,92 @@ title: Installing Calico for the Docker Containerizer
...
-It is very easy to install Calico to use with the
-Docker Containerizer.
+Under the covers, Calico networks Docker tasks for Mesos with its Docker CNM plugin.
+Multihost Networking in Docker requires that
+each Agent's Docker daemon must be configured with a cluster store.
 
-1. On each Mesos Agents, download the `calicoctl` command-line tool:
+Though Docker's configured cluster-store does not have to be the same as Calico's,
+for simplicity, users can configure Docker to use the same by setting the
+following flag in `/etc/docker/daemon.json`:
```
Comment (robbrockbank) [inline master/getting-started/mesos/installation/docker.md]: Do we stipulate a minimum Docker version?   The daemon.json i believe is only valid from 1.10 - prior to 1.10 you have to specify this as the command options.

File: master/getting-started/mesos/installation/docker.md [hunk 7]
```diff
@@ -5,65 +5,92 @@ title: Installing Calico for the Docker Containerizer
...
 
-2. Launch the `calico/node` container.
+Replace `<ETCD_IP>` and `<ETCD_PORT>` with the appropriate `hostname:port` for your etcd cluster.
+
+#### 3. Docker Containerizer Enabled for Mesos Agents
 
-For production deployments, we recommend running the
-container as a service. Visit our guide on [running Calico
-as a service]({{site.baseurl}}/{{page.version}}/usage/configuration/as-service) to learn how to do this.
+By default, Mesos enables only the "Mesos" Containerizer. Ensure
```
Comment (robbrockbank) [inline master/getting-started/mesos/installation/docker.md]: it's a little nicer to say "only enables" imho.

File: master/getting-started/mesos/installation/docker.md [hunk 8]
```diff
@@ -5,65 +5,92 @@ title: Installing Calico for the Docker Containerizer
...
+$ sh -c 'echo docker > /etc/mesos-slave/containerizers'
+$ systemctl restart mesos-slave.service
 ```
 
-Again, be sure to set the ETCD_ENDPOINTS to the correct value for your etcd cluster.
 
-3. Ensure calico's services are running by checking for the Calico container in Docker:
+## Installing Calico
+
+To install Calico, on each Mesos Agent, download the `calicoctl` command-line tool:
```
Comment (robbrockbank) [inline master/getting-started/mesos/installation/docker.md]: I feel that the we either should:
-  Move the warning about this being for trial only being moved to the top
-  or, just say that you need to add a systemd or similar and be done with it.

I'm worried that we'll say run `calicoctl node run` and somewhere down the line someone will think that it's deployed and resilient.   WDYT?   

I'd probably just move the warning to the top.  I have more ideas about what i want the install to look like in the future.

**Comment by robbrockbank** [comment]
Looks fine.  I've got some suggestions, but I wouldn't be upset if you merged as is.

**Pull request merged**